File: experiment/python/gpio2.py
#!/usr/bin/python

import RPi.GPIO as GPIO
import time
import c

delay=0.01

c.init()
c.yellowOff()
c.blueOff()

time.sleep(delay)

diff=1
button=1
consecutive = 0
while consecutive < 400:
  button = GPIO.input(24)
  if button == 0:
    consecutive += diff
    c.blueOn()
  else:
    consecutive = 0
    c.blueOff()

  # If getting closer, we blink with higher frequency.
  if ( consecutive > 200 ):
    diff = 2

  if (consecutive/10) % 2 == 0:
    c.yellowOn()
  else:
    c.yellowOff()

  time.sleep(delay)


c.cleanup()

File: experiment/wiringPi/gpio2.c
#include <wiringPi.h>
#include <unistd.h>

#define BLUE 18
#define YELLOW 17
#define BUTTON 24

#define yellowOn() digitalWrite(YELLOW, HIGH)
#define yellowOff() digitalWrite(YELLOW, LOW)
#define blueOn() digitalWrite(BLUE, HIGH)
#define blueOff() digitalWrite(BLUE, LOW)

void init() {
  wiringPiSetupGpio() ;
  pinMode (BLUE, OUTPUT) ;
  pinMode (YELLOW, OUTPUT) ;
  pinMode (BUTTON, INPUT ) ;
  pullUpDnControl(BUTTON, PUD_UP);
}

void cleanup() {
  yellowOff();
  blueOff();
}



int main (void)
{
  int diff=1;
  int button=1;
  int consecutive=0;

  init();
  while(consecutive<400) {
    button = digitalRead(BUTTON);
    if ( button == LOW ) {
      consecutive += diff;
      blueOn();
    } else {
      consecutive = 0;
      blueOff();
    }
    
    if ( consecutive > 200 )
      diff = 2;

    if ( (consecutive/10) % 2 == 0 ) {
      yellowOn();
    } else {
      yellowOff();
    }
    usleep(10000);
  }
  cleanup();
  return 0 ;
}
